sympy/algebras/quaternion.py

```diff
@@ -742,10 +764,17 @@ def _generic_mul(q1, q2):
             else:
                 raise ValueError("Only commutative expressions can be multiplied with a Quaternion.")
 
+        # If any of the quaternions has a fixed norm, pre-compute norm
+        if q1._norm is None and q2._norm is None:
+            norm = None
```

Comment:
Should that be `or`?

Reply:
It should be `and`! This is to not pre compute the norm if none of the input norms are being pre computed, but to pass on the pre computed norm if one of the input quaternions have.

Reply:
If `q1` has a norm of 1 and `q2` has norm of None, what is `q1.norm() * q2.norm()`?

Reply:
Oh - the norm gets computed! And if they are both None, they remain so and the return-value's norm is also None (not being computed unless needed).

Reply:
> Oh - the norm gets computed! And if they are both None, they remain so and the return-value's norm is also None (not being computed unless needed).

Exactly! :)